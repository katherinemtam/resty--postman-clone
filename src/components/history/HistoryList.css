.HistoryList li {
  background-color: #204060;
  color:white;

  border-radius: 10px;
  padding: 5px;
  margin: 5px;

  font-family: Verdana, Geneva, Tahoma, sans-serif;
  list-style: none;
}

.HistoryList p, .HistoryList h4 {
  margin: 5px;
}

.HistoryList h4 {
  font-weight: bolder;
}


