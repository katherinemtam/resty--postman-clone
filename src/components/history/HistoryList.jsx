import React from 'react';
import PropTypes from 'prop-types';
import History from './History';
import styles from './HistoryList.css';

const HistoryList = ({ history, onClick }) => {
  const historyElements = history.map((eachHistory, index) => (
    <li 
      key={index}
      onClick={onClick}
    >
      <History 
        id={`${eachHistory.search}${eachHistory.method}${eachHistory.body}`}
        search={eachHistory.search}
        method={eachHistory.method}
      />
    </li>
  ));

  if(history === []) {
    return (
      <section className={styles.HistoryList}>
        <h1>HISTORY</h1>
        <ul>Looks like you haven't fetched anything yet.</ul>
      </section>
    );
  } else { 
    return (
      <section className={styles.HistoryList}>
        <h1>HISTORY</h1>
        <ul>{historyElements}</ul>
      </section>
    );
  }
};

HistoryList.protoTypes = {
  history: PropTypes.arrayOf(
    PropTypes.shape({
      search: PropTypes.string.isRequired,
      method: PropTypes.string.isRequired,
    })
  ).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default HistoryList;
