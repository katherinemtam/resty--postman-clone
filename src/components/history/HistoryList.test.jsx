import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  const history = [
    {
      method: 'POST',
      search: 'https://aztro.sameerkumar.website/?sign=pisces&day=today'
    }
  ];

  afterEach(() => cleanup());

  it('renders HistoryList', () => {
    const { asFragment } = render(<HistoryList history={history}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
