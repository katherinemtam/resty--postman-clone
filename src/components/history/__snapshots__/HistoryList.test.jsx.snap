// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<DocumentFragment>
  <section
    class="HistoryList"
  >
    <h1>
      HISTORY
    </h1>
    <ul>
      <li>
        <article
          id="https://aztro.sameerkumar.website/?sign=pisces&day=todayPOSTundefined"
        >
          <h4
            id="https://aztro.sameerkumar.website/?sign=pisces&day=todayPOSTundefined"
          >
            POST
          </h4>
          <p
            id="https://aztro.sameerkumar.website/?sign=pisces&day=todayPOSTundefined"
          >
            https://aztro.sameerkumar.website/?sign=pisces&day=today
          </p>
        </article>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
